sdk/common/src/main/java/io/opentelemetry/sdk/common/config/ConfigProperties.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.common.config;
+
+import static io.opentelemetry.api.internal.ConfigUtil.defaultIfNull;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/** Properties used for configuration of the OpenTelemetry SDK components. */
+public interface ConfigProperties {
```

Comment:
A small variation of this would be to make this version of `ConfigProperties` internal and not expose the `EnvironmentResource#create(ConfigProperties)` method. Would mean you can only access the environment resource using the environment variables and system properties obtains from `System.*` calls - no customizing by providing your own `ConfigProperties` implementation (except in the SPI).

Reply:
That might be a preferable starting point, then if we get requests for customization, we can consider exposing it?  Still thinking through the implications of changing the parent interface on the one in the spi module. 🤔

Reply:
> then if we get requests for customization, we can consider exposing it?

That was my thought.. but on second thought, it may not work. If we move the `:sdk:common` version to an internal package, then the version in `:sdk-extensions:autoconfigure-spi` will extend an internal interface. If we later wanted to change `:sdk:common` to be public facing, I'm pretty sure japicmp would complain.